File: backend/app/models/design_plan.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, JSON
from sqlalchemy.orm import relationship
from app.db.database import Base
import datetime


class DesignPlan(Base):
    """设计方案模型 - AI 生成的美甲设计"""

    __tablename__ = "design_plans"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    customer_id = Column(Integer, ForeignKey("customers.id"), nullable=True, index=True)

    # AI 生成信息
    ai_prompt = Column(Text, nullable=False, comment="用于生成的AI提示词")
    generated_image_path = Column(String(500), nullable=False, comment="生成的设计图路径")
    model_version = Column(String(50), comment="使用的AI模型版本（如 dall-e-3）")

    # 设计参数
    design_target = Column(String(20), comment="设计目标（single/5nails/10nails）")
    style_keywords = Column(
        JSON,
        comment="风格关键词列表（JSON数组）"
    )

    # 参考图片
    reference_images = Column(
        JSON,
        comment="参考图片路径列表（JSON数组）"
    )

    # 版本控制（用于设计迭代）
    parent_design_id = Column(Integer, ForeignKey("design_plans.id"), nullable=True, comment="父设计方案ID（用于追踪迭代）")
    version = Column(Integer, default=1, comment="版本号")
    refinement_instruction = Column(Text, comment="迭代优化指令")

    # AI 估算信息
    estimated_duration = Column(Integer, comment="预估耗时（分钟）")
    estimated_materials = Column(
        JSON,
        comment="预估材料清单（JSON数组）"
    )
    difficulty_level = Column(String(20), comment="难度等级（简单/中等/困难）")

    # 其他信息
    title = Column(String(200), comment="设计方案标题")
    notes = Column(Text, comment="备注")
    is_archived = Column(Integer, default=0, comment="是否归档（1=是，0=否）")

    # 时间戳
    created_at = Column(DateTime, default=datetime.datetime.utcnow, index=True)
    updated_at = Column(DateTime, default=datetime.datetime.utcnow, onupdate=datetime.datetime.utcnow)

    # 关系
    user = relationship("User", back_populates="design_plans")
    customer = relationship("Customer", back_populates="design_plans")
    service_records = relationship("ServiceRecord", back_populates="design_plan")

    # 自引用关系（设计迭代）
    parent_design = relationship("DesignPlan", remote_side=[id], backref="child_designs")

    def __repr__(self):
        return f"<DesignPlan(id={self.id}, title={self.title}, version={self.version})>"

File: frontend/nail_app/lib/models/design_plan.dart
import 'package:json_annotation/json_annotation.dart';

part 'design_plan.g.dart';

/// 设计方案数据模型
@JsonSerializable()
class DesignPlan {
  final int id;

  @JsonKey(name: 'user_id')
  final int userId;

  @JsonKey(name: 'customer_id')
  final int? customerId;

  final String? title;
  final String? notes;

  @JsonKey(name: 'ai_prompt')
  final String aiPrompt;

  @JsonKey(name: 'generated_image_path')
  final String generatedImagePath;

  @JsonKey(name: 'model_version')
  final String? modelVersion;

  @JsonKey(name: 'design_target')
  final String? designTarget;

  @JsonKey(name: 'style_keywords')
  final List<String>? styleKeywords;

  @JsonKey(name: 'reference_images')
  final List<String>? referenceImages;

  @JsonKey(name: 'parent_design_id')
  final int? parentDesignId;

  final int version;

  @JsonKey(name: 'refinement_instruction')
  final String? refinementInstruction;

  @JsonKey(name: 'estimated_duration')
  final int? estimatedDuration;

  @JsonKey(name: 'estimated_materials')
  final List<String>? estimatedMaterials;

  @JsonKey(name: 'difficulty_level')
  final String? difficultyLevel;

  @JsonKey(name: 'is_archived')
  final dynamic isArchived;

  @JsonKey(name: 'created_at')
  final DateTime createdAt;

  @JsonKey(name: 'updated_at')
  final DateTime updatedAt;

  DesignPlan({
    required this.id,
    required this.userId,
    this.customerId,
    this.title,
    this.notes,
    required this.aiPrompt,
    required this.generatedImagePath,
    this.modelVersion,
    this.designTarget,
    this.styleKeywords,
    this.referenceImages,
    this.parentDesignId,
    this.version = 1,
    this.refinementInstruction,
    this.estimatedDuration,
    this.estimatedMaterials,
    this.difficultyLevel,
    this.isArchived = 0,
    required this.createdAt,
    required this.updatedAt,
  });

  bool get archived {
    if (isArchived is bool) return isArchived;
    if (isArchived is int) return isArchived == 1;
    return false;
  }

  factory DesignPlan.fromJson(Map<String, dynamic> json) =>
      _$DesignPlanFromJson(json);
  Map<String, dynamic> toJson() => _$DesignPlanToJson(this);

  @override
  String toString() {
    return 'DesignPlan(id: $id, title: $title, version: $version)';
  }
}

/// 设计方案列表响应
@JsonSerializable()
class DesignPlanListResponse {
  final int total;
  final List<DesignPlan> designs;

  DesignPlanListResponse({
    required this.total,
    required this.designs,
  });

  factory DesignPlanListResponse.fromJson(Map<String, dynamic> json) =>
      _$DesignPlanListResponseFromJson(json);
  Map<String, dynamic> toJson() => _$DesignPlanListResponseToJson(this);
}
